pkg/apis/v1beta1/labels.go

```diff
@@ -114,6 +118,7 @@ var (
 	LabelInstanceAcceleratorName              = Group + "/instance-accelerator-name"
 	LabelInstanceAcceleratorManufacturer      = Group + "/instance-accelerator-manufacturer"
 	LabelInstanceAcceleratorCount             = Group + "/instance-accelerator-count"
+	LabelInstancePerformanceMode              = Group + "/instance-performance-mode"
```

Comment:
```suggestion
	LabelInstancePerformanceMode              = Group + "/instance-capability-flex"
```

Suggesting this label name in the original issue because of the way that EC2 describes their own naming convention here: https://docs.aws.amazon.com/ec2/latest/instancetypes/instance-type-names.html. Given that page as well, I'm not sure that it makes sense to couple the burstable instance types with the flex ones. Can we treat flex as its own thing here?

Reply:
Personally I would prefer a value that can be fetched from the EC2 API, instead of relaying on the instance name structure.
Anyhow, it can be done fairly easy.

Reply:
@jonathan-innis 

I tried to add a label per each instance capability (total 7) and encountered the following error:

`"NodeClaim.karpenter.sh \"spiritslow-94-ywrmgisn0j-p88dm\" is invalid: [spec.requirements: Too many: 31: must have at most 30 items, <nil>: Invalid value: \"null\": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation]"`

The cause is [this line](https://github.com/kubernetes-sigs/karpenter/blob/main/pkg/apis/v1beta1/nodeclaim.go#L40) from kubernetes-sigs/karpenter, can we increase the limit of requirements?